common/lib/dependabot/pull_request_creator/branch_namer/group_rule_strategy.rb

```diff
@@ -0,0 +1,28 @@
+# frozen_string_literal: true
+
+module Dependabot
+  class PullRequestCreator
+    class BranchNamer
+      class GroupRuleStrategy
+        def initialize(dependencies:, files:, target_branch:, group_rule:,
+                       separator: "/", prefix: "dependabot", max_length: nil)
+          @dependencies  = dependencies
+          @files         = files
+          @target_branch = target_branch
+          @group_rule    = group_rule
+          @separator     = separator
+          @prefix        = prefix
+          @max_length    = max_length
+        end
+
+        def new_branch_name
+          group_rule.name
```

Comment:
We should probably expect a group rule name to be a humanized string, as we'll need to either:
- Validate it strictly as being a valid branch name in the config file
- Accept any string and 'slugify' it here replacing invalid characters/spaces etc

I'd tend to prefer the latter as while it might end up making sense to validate it in the config for other reasons, this component probably shouldn't trust its inputs?